pkg/cmd/server/origin/config.go

```diff
@@ -0,0 +1,276 @@
+package origin
+
+import (
+	"crypto/x509"
+	"net/http"
+	"net/url"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/admission"
+	kapi "github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/apiserver"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/auth/user"
+	kclient "github.com/GoogleCloudPlatform/kubernetes/pkg/client"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/tools"
+	kutil "github.com/GoogleCloudPlatform/kubernetes/pkg/util"
+	"github.com/GoogleCloudPlatform/kubernetes/plugin/pkg/admission/admit"
+
+	"github.com/openshift/origin/pkg/api/latest"
+	"github.com/openshift/origin/pkg/auth/authenticator"
+	"github.com/openshift/origin/pkg/auth/authenticator/request/bearertoken"
+	"github.com/openshift/origin/pkg/auth/authenticator/request/paramtoken"
+	"github.com/openshift/origin/pkg/auth/authenticator/request/unionrequest"
+	"github.com/openshift/origin/pkg/auth/authenticator/request/x509request"
+	"github.com/openshift/origin/pkg/auth/group"
+	authnregistry "github.com/openshift/origin/pkg/auth/oauth/registry"
+	"github.com/openshift/origin/pkg/authorization/authorizer"
+	authorizationetcd "github.com/openshift/origin/pkg/authorization/registry/etcd"
+	osclient "github.com/openshift/origin/pkg/client"
+	"github.com/openshift/origin/pkg/cmd/util/variable"
+	oauthetcd "github.com/openshift/origin/pkg/oauth/registry/etcd"
+)
+
+const (
+	unauthenticatedUsername = "system:anonymous"
+
+	authenticatedGroup   = "system:authenticated"
+	unauthenticatedGroup = "system:unauthenticated"
+)
+
+type SystemAddresses struct {
```

Comment:
Wondering if addresses and certs should be sliced by server. Feels a little weird to have multiple config structs spanning the same servers.